pcsx2/SPU2/RegLog.cpp

```diff
@@ -0,0 +1,310 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2020  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+#include "Global.h"
+
+const char* ParamNames[8] = {"VOLL", "VOLR", "PITCH", "ADSR1", "ADSR2", "ENVX", "VOLXL", "VOLXR"};
+const char* AddressNames[6] = {"SSAH", "SSAL", "LSAH", "LSAL", "NAXH", "NAXL"};
+
+__forceinline void _RegLog_(const char* action, int level, const char* RName, u32 mem, u32 core, u16 value)
+{
+	if (level > 1)
+		FileLog("[%10d] SPU2 %s mem %08x (core %d, register %s) value %04x\n",
+				Cycles, action, mem, core, RName, value);
+}
+
+#define RegLog(lev, rname, mem, core, val) _RegLog_(action, lev, rname, mem, core, val)
+
+void SPU2writeLog(const char* action, u32 rmem, u16 value)
```

Comment:
Config, GUI and logging will all be centralized once plugins are merged. I'm letting this be for now to avoid this PR in delay hell, feel free to recomment if you think I should do that for this PR!